### PR TITLE
Task/614 cpp

### PR DESCRIPTION
**Pull request by santosh-kavhar**
https://antaris.atlassian.net/browse/AT-5732

### CODE REVIEW

File: examples/app-cpp/payload_app.cc [hunk 1]
```diff
@@ -395,6 +396,41 @@ void wakeup_seq_fsm(mythreadState_t *threadState)
...
+    printf("Before resp_payload_metrics_params\n");
+
+    time_t now = time(0);
+    UINT64 epoch = static_cast<UINT64>(now);
+    resp_payload_metrics_params.correlation_id = payload_metrics_param->correlation_id;
+    resp_payload_metrics_params.used_counter = 8;   // Example value
+    resp_payload_metrics_params.timestamp = epoch;
+
+    // Total number of counters supported is 8, hence setting all names to 0
+    for (int i = 0; i < 8 ; i++) {
```
[the remark this answers is not in the document]
Comment (santosh-kavhar) [reply examples/app-cpp/payload_app.cc]: Change done

File: examples/app-cpp/payload_app.cc [hunk 2]
```diff
@@ -395,6 +399,41 @@ void wakeup_seq_fsm(mythreadState_t *threadState)
...
+
+    time_t now = time(0);
+    UINT64 epoch = static_cast<UINT64>(now);
+    resp_payload_metrics_params.correlation_id = payload_metrics_param->correlation_id;
+    resp_payload_metrics_params.used_counter = 8;   // Example value
+    resp_payload_metrics_params.timestamp = epoch;
+
+    // Total number of counters supported is 8, hence setting all names to 0
+    for (int i = 0; i < USED_COUNTER_MAX ; i++) {
+        resp_payload_metrics_params.metrics[i].counter = 0;
```
Comment (rahulbhivare) [inline examples/app-cpp/payload_app.cc:416]: What if we change array size in xml file? Do I need to change here as well?

can u use something like:

    size_t metrics_size = sizeof(((struct PayloadMetricsResponse*)0)->metrics) / sizeof(struct PayloadMetricsInfo);

Comment (santosh-kavhar) [reply examples/app-cpp/payload_app.cc:419]: Change Done. XML changes will be automatically reflected here.

File: examples/app-cpp/payload_app.cc [hunk 3]
```diff
@@ -44,6 +45,9 @@
 
 #define STAGE_FILE_DOWNLOAD_DIR         "/opt/antaris/outbound/"    // path for staged file download
 #define STAGE_FILE_NAME                 "SampleFile.txt"            // name of staged file
+
+#define PAYLOAD_METRICS_INFO_NAME_SIZE  16
```
Comment (rahulbhivare) [inline examples/app-cpp/payload_app.cc:49]: Delete this

File: examples/app-cpp/payload_app.cc [hunk 4]
```diff
@@ -395,6 +399,44 @@ void wakeup_seq_fsm(mythreadState_t *threadState)
...
+
+    printf("Before resp_payload_metrics_params\n");
+
+    time_t now = time(0);
+    UINT64 epoch = static_cast<UINT64>(now);
+    resp_payload_metrics_params.correlation_id = payload_metrics_param->correlation_id;
+    resp_payload_metrics_params.used_counter = 8;   // Example value
+    resp_payload_metrics_params.timestamp = epoch;
+
+    size_t used_counter_max = sizeof(((struct PayloadMetricsResponse*)0)->metrics) / sizeof(struct PayloadMetricsInfo);
```
Comment (rahulbhivare) [inline examples/app-cpp/payload_app.cc:414]: change name : 
used_counter_max = max_metrics_counter
name_size = metrics_name

Comment (santosh-kavhar) [reply examples/app-cpp/payload_app.cc]: Done

**Review by rahulbhivare — APPROVED** [review]

**Pull request merged**